### PR TITLE
[issue_tracker] Error given when updating issue

### LINKED ISSUE #7846: [issue_tracker] Error given when updating issue
**Describe the bug**
When I add a comment to an issue and click "Submit" I get the following error swal: 
image

as well as the following in my error_log:
`[Tue Nov 23 10:13:01.880401 2021] [php7:notice] [pid 31784] [client ::1:60308] [ERROR] The server is not configured to receive a file this large., referer: http://localhost:8080/issue_tracker/issue/25`

However, when I refresh, the issue is updated with my new comment. 

I'm not sure what the error in the error log is referring to because there was no file / attachment in the issue. 

**To Reproduce**
Steps to reproduce the behavior (attach screenshots if applicable):
1. Go to issue tracker
2. Click on any issue
3. Enter a new comment
4. Click "Submit"
5. See error

**What did you expect to happen?**
The issue to be successfully updated with a success swal.

### PR DESCRIPTION
**Pull request by maltheism**
## Brief summary of changes

Solves unnecessary error message when making a comment in issue_tracker module without uploading an attachment. Also solves error messages not being displayed to user and solves form value wrongly being cleared causing confusion for user.

#### Testing instructions (if applicable)

1. Checkout PR
2. Make comment in issue tracker module

#### Link(s) to related issue(s)

* Resolves #7846

### CODE REVIEW

File: modules/issue_tracker/php/edit.class.inc [hunk 1]
```diff
@@ -458,7 +458,7 @@ class Edit extends \NDB_Page implements ETagCalculator
 
         // Attachment for new issue.
         $uploadedFiles = $request->getUploadedFiles();
-        if (isset($uploadedFiles)) {
+        if (isset($uploadedFiles) && !empty($uploadedFiles)) {
```
Comment (driusan) [inline modules/issue_tracker/php/edit.class.inc:461]:
```suggestion
        if (!empty($uploadedFiles)) {
```

the isset is unnecessary. empty already suppresses unset errors. (And it's always set, it's declared on the line above)

**Comment by maltheism** [comment]
thanks @driusan

**Review by CamilleBeau — APPROVED** [review]

File: modules/issue_tracker/jsx/IssueForm.js [hunk 2]
```diff
@@ -402,18 +402,19 @@ class IssueForm extends Component {
   }
 
   /**
-   * Handles form submission
+   * Handles form submission for new issue being created
```
Comment (maltheism) [inline modules/issue_tracker/jsx/IssueForm.js:405]: Updated comment

File: modules/issue_tracker/jsx/IssueForm.js [hunk 3]
```diff
@@ -422,6 +423,9 @@ class IssueForm extends Component {
       return;
     }
 
+    // Prevent multiple submissions
```
Comment (maltheism) [inline modules/issue_tracker/jsx/IssueForm.js:426]: cleanup

File: modules/issue_tracker/jsx/IssueForm.js [hunk 4]
```diff
@@ -441,7 +445,7 @@ class IssueForm extends Component {
         response.json().then((data) => {
           this.setState({submissionResult: 'error'});
           let msgType = 'error';
-          let message = data.message || 'Failed to submit issue :(';
+          const message = data.error ?? data.message;
```
Comment (maltheism) [inline modules/issue_tracker/jsx/IssueForm.js:448]: This change makes it so the user actually sees the error message if one happens. Example visit needs a PSCID when submitting issue with visit.

File: modules/issue_tracker/php/edit.class.inc [hunk 5]
```diff
@@ -433,6 +433,9 @@ class Edit extends \NDB_Page implements ETagCalculator
         $issueValues['lastUpdatedBy'] = $user->getUsername();
 
         $validatedInput = $this->validateInput($validateValues, $user);
+        if (!is_array($validatedInput)) { // Error exists.
```
Comment (maltheism) [inline modules/issue_tracker/php/edit.class.inc:436]: This change fixes error message not being returned.

File: modules/issue_tracker/php/edit.class.inc [hunk 6]
```diff
@@ -473,7 +476,7 @@ class Edit extends \NDB_Page implements ETagCalculator
                     \Psr\Log\LogLevel::ERROR,
                     $attachment->errorMessage
                 );
-                return new \LORIS\Http\Response\JSON\InternalServerError(
+                return new \LORIS\Http\Response\JSON\Conflict(
```
Comment (maltheism) [inline modules/issue_tracker/php/edit.class.inc:479]: I think Conflict makes more sense here.

File: modules/issue_tracker/php/edit.class.inc [hunk 7]
```diff
@@ -682,10 +685,10 @@ class Edit extends \NDB_Page implements ETagCalculator
             }
             if (!$validCenter) {
                 $this->logger->log(
-                    \Psr\Log\LogLevel::ERROR,
+                    \Psr\Log\LogLevel::WARNING,
                     'PSCID and Center ID do not match a valid session!'
                 );
-                return new \LORIS\Http\Response\JSON\InternalServerError(
+                return new \LORIS\Http\Response\JSON\Conflict(
```
Comment (maltheism) [inline modules/issue_tracker/php/edit.class.inc:691]: I think Conflict makes more sense here.

File: modules/issue_tracker/php/edit.class.inc [hunk 8]
```diff
@@ -710,7 +713,7 @@ class Edit extends \NDB_Page implements ETagCalculator
                     \Psr\Log\LogLevel::ERROR,
                     'PSCID and Visit Label do not match a valid candidate session!'
                 );
-                return new \LORIS\Http\Response\JSON\InternalServerError(
+                return new \LORIS\Http\Response\JSON\Conflict(
```
Comment (maltheism) [inline modules/issue_tracker/php/edit.class.inc:716]: I think Conflict makes more sense here.

File: modules/issue_tracker/php/edit.class.inc [hunk 9]
```diff
@@ -735,7 +738,7 @@ class Edit extends \NDB_Page implements ETagCalculator
                     \Psr\Log\LogLevel::ERROR,
                     'PSCID does not match a valid candidate!'
                 );
-                return new \LORIS\Http\Response\JSON\InternalServerError(
+                return new \LORIS\Http\Response\JSON\Conflict(
```
Comment (maltheism) [inline modules/issue_tracker/php/edit.class.inc:741]: I think Conflict makes more sense here.

File: modules/issue_tracker/php/edit.class.inc [hunk 10]
```diff
@@ -749,7 +752,7 @@ class Edit extends \NDB_Page implements ETagCalculator
                 \Psr\Log\LogLevel::ERROR,
                 'Visit Label must be accompanied by a PSCID'
             );
-            return new \LORIS\Http\Response\JSON\InternalServerError(
+            return new \LORIS\Http\Response\JSON\Conflict(
```
Comment (maltheism) [inline modules/issue_tracker/php/edit.class.inc:755]: I think Conflict makes more sense here.

**Comment by maltheism** [comment]
Hi @SantiagoTG please review this PR because I added changes to solve the issue #7884.

File: modules/issue_tracker/php/edit.class.inc [hunk 11]
```diff
@@ -565,9 +567,14 @@ class Edit extends \NDB_Page implements ETagCalculator
 
         if (isset($changed_assignee)) {
             $issueChangeEmailsAssignee = $db->pselect(
-                "SELECT u.Email as Email, u.First_name as firstname " .
-                "FROM users u WHERE u.UserID=:assignee
-                AND u.UserID<>:currentUser",
+                "SELECT
```
Comment (maltheism) [inline modules/issue_tracker/php/edit.class.inc:570]: Fixed query for sending email here.

File: modules/issue_tracker/php/edit.class.inc [hunk 12]
```diff
@@ -594,11 +601,11 @@ class Edit extends \NDB_Page implements ETagCalculator
                         u.First_name AS firstname
                     FROM
                         users u
-                        INNER JOIN issues_watching w ON (w.userID = u.userID)
+                        INNER JOIN issues_watching w ON (w.userID = u.UserID)
                     WHERE
                         w.issueID = :issueID
-                        AND u.UserID <> :uid
-                        AND u.UserID <> :assignee",
+                        AND w.userID = :uid
```
Comment (maltheism) [inline modules/issue_tracker/php/edit.class.inc:607]: Fixed query for sending email here.

**Comment by maltheism** [comment]
Hi @kongtiaowang can you test the sending email part of this PR as well? My development setup isn't receiving emails for some reason I'm having trouble figuring out and while the code is now running correctly. I tested receiving emails for request account that used to work but I'm still not receiving emails. Issue 14 should send an email to the admin account when a comment is made. The issues_watching table in the database has other issues that will send emails.

**Pull request merged**